packages/twenty-front/src/modules/object-record/object-options-dropdown/components/ObjectOptionsDropdownRecordGroupsContent.tsx

```diff
@@ -12,56 +12,72 @@ import {
 import { hiddenRecordGroupIdsComponentSelector } from '@/object-record/record-group/states/selectors/hiddenRecordGroupIdsComponentSelector';
-import { visibleRecordGroupIdsComponentSelector } from '@/object-record/record-group/states/selectors/visibleRecordGroupIdsComponentSelector';
-import { recordIndexRecordGroupHideComponentState } from '@/object-record/record-index/states/recordIndexRecordGroupHideComponentState';
-import { recordIndexRecordGroupIsDraggableSortComponentSelector } from '@/object-record/record-index/states/selectors/recordIndexRecordGroupIsDraggableSortComponentSelector';
+import { visibleRecordGroupIdsComponentFamilySelector } from '@/object-record/record-group/states/selectors/visibleRecordGroupIdsComponentFamilySelector';
+import { recordIndexRecordGroupHideComponentFamilyState } from '@/object-record/record-index/states/recordIndexRecordGroupHideComponentFamilyState';
+import { recordIndexRecordGroupSortComponentState } from '@/object-record/record-index/states/recordIndexRecordGroupSortComponentState';
 import { DropdownMenuHeader } from '@/ui/layout/dropdown/components/DropdownMenuHeader';
 import { DropdownMenuItemsContainer } from '@/ui/layout/dropdown/components/DropdownMenuItemsContainer';
 import { DropdownMenuSeparator } from '@/ui/layout/dropdown/components/DropdownMenuSeparator';
+import { useRecoilComponentFamilyValueV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentFamilyValueV2';
 import { useRecoilComponentValueV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentValueV2';
+import { useGetCurrentView } from '@/views/hooks/useGetCurrentView';
 import { useIsFeatureEnabled } from '@/workspace/hooks/useIsFeatureEnabled';
 
 export const ObjectOptionsDropdownRecordGroupsContent = () => {
   const isViewGroupEnabled = useIsFeatureEnabled('IS_VIEW_GROUPS_ENABLED');
 
-  const { currentContentId, recordIndexId, onContentChange, resetContent } =
-    useOptionsDropdown();
+  const {
+    viewType,
+    currentContentId,
+    recordIndexId,
+    onContentChange,
+    resetContent,
+  } = useOptionsDropdown();
+
+  const { currentViewWithCombinedFiltersAndSorts: currentView } =
+    useGetCurrentView();
 
   const recordGroupFieldMetadata = useRecoilComponentValueV2(
     recordGroupFieldMetadataComponentState,
   );
 
-  const visibleRecordGroupIds = useRecoilComponentValueV2(
-    visibleRecordGroupIdsComponentSelector,
+  const visibleRecordGroupIds = useRecoilComponentFamilyValueV2(
+    visibleRecordGroupIdsComponentFamilySelector,
+    viewType,
   );
 
   const hiddenRecordGroupIds = useRecoilComponentValueV2(
     hiddenRecordGroupIdsComponentSelector,
   );
 
-  const isDragableSortRecordGroup = useRecoilComponentValueV2(
-    recordIndexRecordGroupIsDraggableSortComponentSelector,
+  const hideEmptyRecordGroup = useRecoilComponentFamilyValueV2(
```

Comment:
just a thought - are we sure we want to hide the empty groups by default? I would have done the opposite, otherwise it can feel like they are just missing (that was my impression)
@Bonapara

Reply:
@ijreilly It's only hidden by default on the table view

Reply:
Yes but it means you have to create a new view to access it. I was just pointing out that I wasnt sure that is what was intended, i will let thomas clarify !

Reply:
It was actually requested by @Bonapara 🙂

Reply:
I said that since Notion hides empty groups by default, I think it's a safe behavior that prevents clutter, but this can change in the future!

Reply:
I realize my second comment was meant for the above discussion about making the groups by feature only accessible in newly created views, I don't know how I ended up mixing conversations sorry